lightly/api/api_workflow_client.py

```diff
@@ -291,11 +291,16 @@ def set_api_client_request_timeout(
 
     """
     request_fn = client.rest_client.request
+    client.rest_client.request = _request_with_timeout(timeout=timeout, request_fn=request_fn)
 
-    def new_request_fn(*args, **kwargs):
+
+class _request_with_timeout:
```

Comment:
Had to refactor the `new_request_fn` function into a global object to make it picklable.